intelmq/tests/bots/collectors/tcp/test_collector.py

```diff
@@ -0,0 +1,184 @@
+            "time.observation": "2018-03-20T14:05:03+00:00"}
+
+INPUT1 = {"classification.taxonomy": "malicious code",
+          "classification.type": "c&c",
+          "feed.name": "Example feed",
+          "feed.accuracy": 100.,
+          "feed.url": "http://localhost/two_files.tar.gz",
+          "raw": utils.base64_encode('bar text\n'),
+          "time.observation": "2018-03-20T14:05:03+00:00"
+          }
+REPORT1 = {
+    'raw': 'eyJjbGFzc2lmaWNhdGlvbi50YXhvbm9teSI6ICJtYWxpY2lvdXMgY29kZSIsICJjbGFzc2lmaWNhdGlvbi50eXBlIjogImMmYyIsICJmZWVkLm5hbWUiOiAiRXhhbXBsZSBmZWVkIiwgImZlZWQuYWNjdXJhY3kiOiAxMDAuMCwgImZlZWQudXJsIjogImh0dHA6Ly9sb2NhbGhvc3QvdHdvX2ZpbGVzLnRhci5neiIsICJyYXciOiAiWW1GeUlIUmxlSFFLIiwgInRpbWUub2JzZXJ2YXRpb24iOiAiMjAxOC0wMy0yMFQxNDowNTowMyswMDowMCJ9',
+    'feed.name': 'Example feed', 'feed.accuracy': 100.0}
+
+INPUT2 = {
+    "feed.name": "Example feed",
+    "feed.accuracy": 100.,
+    "feed.url": "http://localhost/two_files.tar.gz",
+    "raw": utils.base64_encode('foo text\n'),
+}
+
+
+@test.skip_local_web()
+class TestIntelMQCollectorBot(test.BotTestCase, unittest.TestCase):
+    """
+    A TestCase for HTTPCollectorBot.
+    """
+
+    @classmethod
+    def set_bot(cls):
+        cls.PORT = 5011
+        cls.bot_reference = TCPCollectorBot
+        cls.sysconfig = {'http_url': 'http://localhost/two_files.tar.gz',
+                         'extract_files': True,
+                         'feed': 'Example feed',
+                         'ip': "localhost",
+                         "port": cls.PORT,
+                         "separator": "\n"
+                         }
+        cls.separator = b"\n"
+
+    def _client(self):
+        sleep(1)
+        connection = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        connection.connect(("localhost", self.PORT))
+        return connection
+
+    def _test_collecting_single_message(self):
+        """ Test if correct Events have been produced. """
+
+        def client():
+            data = Event(INPUT1).to_json(hierarchical=False)
+            text = utils.encode(data) + self.separator
+
+            connection = self._client()
+            connection.sendall(text)
+            connection.close()
+
+        thread = threading.Thread(target=client)
+        thread.start()
+        self.input_message = None
+        self.run_bot()
+
+        self.assertOutputQueueLen(1)
+        line = self.get_output_queue()[0]
+        generated_report = MessageFactory.unserialize(line, harmonization=self.harmonization, default_type='Event')
+        original_message = MessageFactory.unserialize(base64_decode(generated_report["raw"]), harmonization=self.harmonization,
+                                               default_type='Event')
+        self.assertDictEqual(original_message, INPUT1)
+        del generated_report["time.observation"]
+        self.assertDictEqual(generated_report, REPORT1)
```

Comment:
You can use `assertMessageEqual` here, it ignores the observation time.

And I get an error here:

```
Traceback (most recent call last):
  File "/home/sebastian/dev/intelmq/intelmq/tests/bots/collectors/tcp/test_collector.py", line 106, in test_collecting_single_message
    self.assertDictEqual(generated_report, REPORT1)
nose.proxy.AssertionError: {'fee[138 chars]WVkLmFjY3VyYWN5IjogMTAwLjAsICJmZWVkLm5hbWUiOiA[202 chars]00.0} != {'fee[138 chars]WVkLm5hbWUiOiAiRXhhbXBsZSBmZWVkIiwgImZlZWQuYWN[202 chars]00.0}
  {'feed.accuracy': 100.0,
   'feed.name': 'Example feed',
-  'raw': 'eyJjbGFzc2lmaWNhdGlvbi50YXhvbm9teSI6ICJtYWxpY2lvdXMgY29kZSIsICJjbGFzc2lmaWNhdGlvbi50eXBlIjogImMmYyIsICJmZWVkLmFjY3VyYWN5IjogMTAwLjAsICJmZWVkLm5hbWUiOiAiRXhhbXBsZSBmZWVkIiwgInJhdyI6ICJZbUZ5SUhSbGVIUUsiLCAidGltZS5vYnNlcnZhdGlvbiI6ICIyMDE4LTAzLTIwVDE0OjA1OjAzKzAwOjAwIiwgImZlZWQudXJsIjogImh0dHA6Ly9sb2NhbGhvc3QvdHdvX2ZpbGVzLnRhci5neiJ9'}
+  'raw': 'eyJjbGFzc2lmaWNhdGlvbi50YXhvbm9teSI6ICJtYWxpY2lvdXMgY29kZSIsICJjbGFzc2lmaWNhdGlvbi50eXBlIjogImMmYyIsICJmZWVkLm5hbWUiOiAiRXhhbXBsZSBmZWVkIiwgImZlZWQuYWNjdXJhY3kiOiAxMDAuMCwgImZlZWQudXJsIjogImh0dHA6Ly9sb2NhbGhvc3QvdHdvX2ZpbGVzLnRhci5neiIsICJyYXciOiAiWW1GeUlIUmxlSFFLIiwgInRpbWUub2JzZXJ2YXRpb24iOiAiMjAxOC0wMy0yMFQxNDowNTowMyswMDowMCJ9'}
```
Because the dict in raw can be sorted differently. The order of elements in dicts is not fixed. So sort the keys in serialization.